src/Microsoft.ML.Transforms/KeyToVectorMapping.cs

```diff
@@ -478,7 +478,7 @@ public override SchemaShape GetOutputSchema(SchemaShape inputSchema)
             {
                 if (!inputSchema.TryFindColumn(colInfo.Input, out var col))
                     throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", colInfo.Input);
-                if ((col.ItemType.ItemType.RawKind == default) || !(col.ItemType.IsVector || col.ItemType is PrimitiveType))
+                if (!(col.ItemType.IsVector || col.ItemType is PrimitiveType))
```

Comment:
>ItemType [](start = 51, length = 8)

I think this is probably going to change anyway with your other change on vectors, but just one thing to keep in mind: While `ItemType` on `ColumnType` itself was a `ColumnType` (which is correct), `ItemType` on the vector type itself is and always has been a `PrimitiveType` anyway. So this test here seems at least odd

Reply:
Actually this is a bug (or at least confusion) in our API in my opinion.  `col` is of type `SchemaShape.Column`. It has a member on it - `public readonly ColumnType ItemType;`. So `col.ItemType` isn't actually the `VectorType.ItemType`. It could be _any_ `ColumnType` (for example, it could be an `ImageType`).

We then split out the "vector"-ness to a different `public readonly VectorKind Kind;` property.

---
In reply to: [248034844](https://github.com/dotnet/machinelearning/pull/2143#discussion_r248034844) [](ancestors = 248034844)